sdk-rust/src/lib.rs

```diff
@@ -109,14 +109,20 @@ pub fn try_from_mem<TOutput: for<'a> Deserialize<'a>>(ptr: *mut u8) -> Result<Op
 
     let json_string = String::from_utf8(input_vec.to_vec())?.clone();
 
-    mem::drop(input_vec);
-    mem::drop(len_rdr);
-    mem::drop(type_rdr);
-
     // It is possible for null json values to reach this line, particularly if sourced directly
     // from a 3rd party module, so we ensure that we parse to option as well as the earlier type_id
     // checks.
     let result = serde_json::from_str::<Option<TOutput>>(&json_string)?;
+
+    // Now that we have finished working in this func, we can construct the full input buffer
+    // and then manually drop it.
+    let buf_len = mem::size_of::<i8>() + mem::size_of::<u32>() + len;
+    let buf: Vec<u8> = unsafe {
+        Vec::from_raw_parts(ptr, buf_len, buf_len)
+    };
+    let buf = ManuallyDrop::new(buf);
+    mem::drop(ManuallyDrop::into_inner(buf));
```

Comment:
question: From what I understand, when dropping something that has been wrapped with `ManuallyDrop` the recommended way is to use:
```go
unsafe {
  ManuallyDrop::drop(buf)
}
```

question:  Why construct the buffer if it is then dropped without being used?

Reply:
> From what I understand, when dropping something that has been wrapped with ManuallyDrop the recommended way is to us

Where did you hear that? If ManuallyDrop was truely unsafe the ManuallyDrop:new() would be marked as unsafe and/or the docs would recommend wrapping with unsafe.

>  Why construct the buffer if it is then dropped without being used
It is components are used further up in the function.  It's length is unknown until it has started to be read from however, and it is easier and somewhat safer to drop it here in one go instead of dropping its components one by one.

Reply:
> Where did you hear that? If ManuallyDrop was truely unsafe the ManuallyDrop:drop() would be marked as unsafe and/or the docs would recommend wrapping with unsafe.

Might have been because I'm used to seeing it used on struct fields and not on variables directly. Usually it's done to manipulate the order of fields being dropped when that is important. If you forget the unsafe, why not `ManuallyDrop::drop(buf)`?

> It is components are used further up in the function. It's length is unknown until it has started to be read from however, and it is easier and somewhat safer to drop it here in one go instead of dropping its components one by one (this also matches how it was allocated).

I still don't understand why you create this buffer but never use it.

question: Why do you no longer have to drop the other 3 variable? They are wrapped in `ManuallyDrop` so if you forgo explicitly dropping them, it feels like it would cause a memory leak.

Reply:
> If you forget the unsafe, why not ManuallyDrop::drop(buf)

`ManuallyDrop::drop(buf)` is the bug caught by the new linter - it is why this PR exists.  The original only dropped the ManuallyDrop wrapper to the buf, not the buf itself.

> Why do you no longer have to drop the other 3 variable

They are included in the `buf`.  `buf` is a block of memory containing the other 3.   Dropping buf forgets all 4 variables and would call `buf`s destructor (if it existed).

For example if you were to use `mem::forget` (which lacks the double free protections of `ManuallyDrop::drop`) you would essentially be forgetting the same blocks of memory twice - once for the 3 components, and then again for `buf`: https://doc.rust-lang.org/std/mem/fn.forget.html#relationship-with-manuallydrop

Reply:
> ManuallyDrop::drop(buf) is the bug caught by the new linter - it is why this PR exists. The original only dropped the ManuallyDrop wrapper to the buf, not the buf itself.

Is there something this PR doesn't show? The diff shows you were using `mem::drop` and not `ManuallyDrop::drop`

Reply:
> Is there something this PR doesn't show? The diff shows you were using mem::drop and not ManuallyDrop::drop

I meant `mem::drop`, I'll correct my comment